testlib/src/main/java/com/diffplug/spotless/StepHarness.java

```diff
@@ -58,14 +59,14 @@ public static StepHarness forFormatter(Formatter formatter) {
 	/** Asserts that the given element is transformed as expected, and that the result is idempotent. */
 	public StepHarness test(String before, String after) {
 		String actual = formatter.compute(LineEnding.toUnix(before), new File(""));
-		assertEquals(after, actual, "Step application failed");
```

Comment:
This gets used for all kinds of tests, not only json tests. Also, it should be *exactly* the same as the snapshot, if it isn't, there's a bug in the formatter.